spec/models/has_two_files_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+
+describe :has_two_uploadcare_files, :vcr do
+  let(:post) { PostWithTwoFiles.new(title: 'Post title', file_1: FILE_1_CDN_URL, file_2: FILE_2_CDN_URL) }
+  let(:subject) { post }
+
+  describe 'object with uploadcare_files' do
+    it 'creates blank post' do
+      PostWithTwoFiles.create!
+    end
+
+    it 'responds to has_uploadcare_file? method' do
+      is_expected.to respond_to(:has_file_1_as_uploadcare_file?)
+      is_expected.to respond_to(:has_file_2_as_uploadcare_file?)
+    end
+
+    it 'responds to has_uploadcare_group? method' do
+      is_expected.to respond_to(:has_file_1_as_uploadcare_group?)
+      is_expected.to respond_to(:has_file_2_as_uploadcare_group?)
+    end
+
+    it 'has Uploadcare::Rails::File' do
+      expect(post.file_1).to be_an(Uploadcare::Rails::File)
+      expect(post.file_2).to be_an(Uploadcare::Rails::File)
+    end
+
+    it 'stores file after save' do
```

Comment:
You also need to check that either a store request was made or file was not stored before `post.save`. Same for 'stores group after save', same for file/group deletion